tables/vpc_flow_log/vpc_flow_log_table.go

```diff
@@ -1,138 +1,365 @@
+				Description: "The ARN of the ECS cluster if traffic originates from an ECS task.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_cluster_name",
+				Description: "The name of the ECS cluster if traffic originates from an ECS task.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_container_instance_arn",
+				Description: "The ARN of the ECS container instance if traffic is from a running ECS task on an EC2 instance.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_container_instance_id",
+				Description: "The ID of the ECS container instance if traffic is from a running ECS task on an EC2 instance.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_container_id",
+				Description: "The Docker runtime ID of the first container in an ECS task.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_second_container_id",
+				Description: "The Docker runtime ID of the second container in an ECS task, if applicable.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_service_name",
+				Description: "The name of the ECS service if traffic is from an ECS task started by a service.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_task_arn",
+				Description: "The ARN of the ECS task if traffic is from a running ECS task.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_task_definition_arn",
+				Description: "The ARN of the ECS task definition if traffic is from a running ECS task.",
+				Type:        "varchar",
+			},
+			{
+				ColumnName:  "ecs_task_id",
+				Description: "The ID of the ECS task if traffic is from a running ECS task.",
+				Type:        "varchar",
+			},
+			// version 8 fields
+			{
+				ColumnName:  "reject_reason",
+				Description: "The reason why traffic was rejected (e.g., BPA).",
+				Type:        "varchar",
+			},
+		},
+		NullValue:   VpcFlowLogTableNilValue,
+		Description: c.GetDescription(),
+	}
+}
+
+func (c *VpcFlowLogTable) GetSourceMetadata() ([]*table.SourceMetadata[*types.DynamicRow], error) {
 
 	defaultS3ArtifactConfig := &artifact_source_config.ArtifactSourceConfigImpl{
 		FileLayout: utils.ToStringPointer("AWSLogs/(%{DATA:org_id}/)?%{NUMBER:account_id}/vpcflowlogs/%{DATA:region}/%{YEAR:year}/%{MONTHNUM:month}/%{MONTHDAY:day}/(%{NUMBER:hour}/)?%{DATA}.log.gz"),
 	}
 
-	format := NewVPCFlowLogTableFormat()
-
-	// // ask our CustomTableImpl for the mapper
-	// mapper, err := c.Format.GetMapper()
-	// if err != nil {
-	// 	return nil, err
-	// }
-
-	slog.Error("mapper ===>>> ", "regex", format.GetProperties())
+	// ask our CustomTableImpl for the mapper
+	mapper, err := c.Format.GetMapper()
+	if err != nil {
+		return nil, err
+	}
 
-	return []*table.SourceMetadata[*VpcFlowLog]{
+	return []*table.SourceMetadata[*types.DynamicRow]{
 		{
 			// S3 artifact source
 			SourceName: s3_bucket.AwsS3BucketSourceIdentifier,
+			Mapper:     &VPCFlowLogMapper{},
```

Comment:
Should use format from mapper?

Reply:
Removed the mapper since it's now fully handled within the extractor, which also takes care of using the provided format.